src/core/providers/canvas-schema/canvas-schema.provider.tsx

```diff
@@ -41,6 +42,14 @@ export const CanvasSchemaProvider: React.FC<Props> = props => {
     );
   };
 
+  const addRelation = (relation: RelationVm) => {
+    setSchema(prevSchema =>
+      produce(prevSchema, draft => {
+        draft.relations.push(relation);
+      })
```

Comment:
I'm going to open a separate case for this in addRelation avoid addin a relation that already exists:

https://github.com/Lemoncode/mongo-modeler/issues/90

Somebody will implement it

Reply:
Added comment for the TODO task